geps/gep-91.md

```diff
@@ -16,7 +16,71 @@ This GEP proposes a way to validate the TLS certificate presented by the downstr
 ## Non-Goals
 - Define other fields that can be used to verify the client certificate such as the Cerificate Hash or Subject Alt Name. 
 
+### API
+
+* Introduce a `clientValidation` field with [Gateway.Tls][] that can be used to validate the client intiating the TLS connection
+to the Gateway
+* Introduce a `caCerficateRefs` field within `clientValidation` that can be used to specify a list of CA Certificates that
+can be used as a trusted anchor to validate the certificates presented by the client
+
+#### GO
+
+```go
+// ClientValidationContext holds configuration that can be used to validate the client intiating the TLS connection
+// to the Gateway.
+// By default, no client specific configuration is validated.
+type ClientValidationContext struct {
```

Comment:
I think here the `ClientValidationContext` should claim some unmentioned behaviors:

1. What if the `caCertificateRefs` field is empty? It's implementation-specific, or all the clients will be rejected, or passed?
2. What's the verify depth of the client certificate? Only the server certificate and the intermediate CA, or the whole chain (Maybe we can add another field like `VerifyDepth`)?

Reply:
Great questions - my thoughts:

1. We could copy the pattern we used for BackendTLSPolicy here, and have something that suggests whether or not to use the system CAs.
2. I think that unless we have a _very_ good reason, we should start with mandating the behavior that the full chain is checked, and then we can add configurability later if we need it. We should make the default case as secure as possible, and not have easily-accessible knobs that make things less secure unless we're _sure_ we need them.

Reply:
@youngnick Fair enough, but we should clarify them in the GEP to reduce the ambiguity

Reply:
I've add a doc string, mentioning that this is implementation specific today, because im seeing different default verifyDepth settings in different controller